packages/discord.js/src/managers/GuildBanManager.js

```diff
@@ -150,7 +153,12 @@ class GuildBanManager extends CachedManager {
     const id = this.client.users.resolveId(user);
     if (!id) throw new Error(ErrorCodes.BanResolveId, true);
     await this.client.rest.put(Routes.guildBan(this.guild.id, id), {
-      body: { delete_message_days: options.deleteMessageDays },
+      body: {
+        delete_message_seconds:
+          typeof options.deleteMessageSeconds !== 'undefined'
+            ? options.deleteMessageSeconds
+            : (options.deleteMessageDays ?? 0) * 24 * 60 * 60,
```

Comment:
Isn't this optional? If it is, we should respect that

Reply:
I think it's still optional, if you don't pass any argument to the banOptions `options.deleteMessageSeconds !== 'undefined'` will evaluate to false, and `options.deleteMessageDays ?? 0` to 0. Unless I'm missing something.

Reply:
In the case you described, yes 0 is the result. That means upon providing nothing, `delete_message_seconds` will be set to `0` and sent to Discord. `delete_message_seconds` should be nothing (`undefined`) - not a set value by discord.js.

Reply:
it defaults to 0 so it doesn't matter. anyway we could change it although it doesn't make a difference; the typeof check seems to be useless tho. https://discord.com/channels/222078108977594368/222197033908436994/1015418401503580271

Reply:
Yes, please change it, hence this requested change in the first place.